src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -38,36 +36,30 @@
 @ComponentScan(basePackageClasses = StudyService.class)
 public class StudyController {
 
-    @Autowired
-    private StudyService studyService;
+    private final StudyService studyService;
+
+    public StudyController(StudyService studyService) {
+        this.studyService = studyService;
+    }
 
     @GetMapping(value = "/studies")
     @ApiOperation(value = "Get all studies")
     @ApiResponses(value = {@ApiResponse(code = 200, message = "The list of studies")})
-    public ResponseEntity<List<StudyInfos>> getStudyList() {
-        List<StudyInfos> studies = studyService.getStudyList();
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studies);
+    public Mono<ResponseEntity<Flux<StudyInfos>>> getStudyList() {
+        Flux<StudyInfos> studies = studyService.getStudyList();
+        return Mono.just(ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(studies));
```

Comment:
Use Flux.collect or a variant to make a chain from start to the end

Reply:
You can change the return type from Mono<ResponseEntity<Flux<StudyInfos>>>> to Mono<ResponseEntity<List<StudyInfos>>>>